server/src/controllers/job_lead/getAllJobLeads.js

```diff
@@ -1,14 +1,138 @@
 const logger = require("pino")();
+const { Op, literal, Sequelize } = require("sequelize");
 const JobLead = require("../../models/job_lead.model");
 
 const getAllJobLeadsRequestHandler = async (req, res) => {
   try {
-    const job_leads = await JobLead.findAll();
+    const page = req?.query?.page ? parseInt(req.query.page, 10) : 0;
+    const pageSize = req?.query?.page ? parseInt(req.query.pageSize, 10) : 10;
+    const offsetSize = page * pageSize;
+
+    const {
+      searchTitleQuery,
+      startDateCreated,
+      endDateCreated,
+      startDateExpired,
+      endDateExpired,
+      minCompensation,
+      maxCompensation,
+      minHoursPerWeek,
+      maxHoursPerWeek,
+      ownerId,
+      searchNOCQuery,
+      jobTypes,
+    } = req.query;
+
+    const query = {};
+    if (searchTitleQuery) {
+      query.job_title = { [Op.like]: `%${searchTitleQuery}%` };
+    }
+    if (startDateCreated) {
+      query.creation_date = { [Op.gte]: new Date(startDateCreated) };
+    }
+
+    if (endDateCreated) {
+      query.creation_date = {
+        ...query.creation_date,
+        [Op.lte]: new Date(endDateCreated),
+      };
+    }
```

Comment:
Currently date filtering is not working as intended (try limiting your search to a single date to reproduce). This might have to do with how the dates are stored in the database... not too sure though.

Reply:
Hey Daniel, good catch, I accidentally swapped the date create and date expired filters logic in the frontend but should be good now. Sorry about that.

Reply:
Hey Kevin, it seems the issue is still persisting.
<img width="1740" alt="Screenshot 2024-01-21 at 8 48 17 PM" src="https://github.com/uoftblueprint/tces/assets/58835213/b2a948cd-8e94-428c-b649-0f1de33de65f">

Reply:
Ohh, sorry I thought the issue was more so the filter, I will look into that rn.

Reply:
Hey Daniel, the issue should be fixed now, the solution was making sure I set the start and end hour of the date query.

<img width="1434" alt="image" src="https://github.com/uoftblueprint/tces/assets/74465053/8abba75a-0102-4962-976a-dcb6c51847bb">